packages/plugin-ext/src/common/known-commands.ts

```diff
@@ -0,0 +1,315 @@
+    mappings['editor.action.smartSelect.shrink'] = ['monaco.editor.action.smartSelect.shrink', createConversionFunction(...[])];
+    mappings['editor.action.forceRetokenize'] = ['monaco.editor.action.forceRetokenize', createConversionFunction(...[])];
+    mappings['editor.action.toggleTabFocusMode'] = ['monaco.editor.action.toggleTabFocusMode', createConversionFunction(...[])];
+    mappings['editor.action.gotoLine'] = ['monaco.editor.action.gotoLine', createConversionFunction(...[])];
+    mappings['editor.action.quickOutline'] = ['monaco.editor.action.quickOutline', createConversionFunction(...[])];
+    mappings['editor.action.inPlaceReplace.up'] = ['monaco.editor.action.inPlaceReplace.up', createConversionFunction(...[])];
+    mappings['editor.action.inPlaceReplace.down'] = ['monaco.editor.action.inPlaceReplace.down', createConversionFunction(...[])];
+    mappings['editor.action.diffReview.next'] = ['monaco.editor.action.diffReview.next', createConversionFunction(...[])];
+    mappings['editor.action.diffReview.prev'] = ['monaco.editor.action.diffReview.prev', createConversionFunction(...[])];
+    mappings['editor.action.selectToBracket'] = ['monaco.editor.action.selectToBracket', createConversionFunction(...[])];
+    mappings['editor.action.jumpToBracket'] = ['monaco.editor.action.jumpToBracket', createConversionFunction(...[])];
+    mappings['actions.findWithSelection'] = ['monaco.actions.findWithSelection', createConversionFunction(...[])];
+    mappings['editor.action.nextMatchFindAction'] = ['monaco.editor.action.nextMatchFindAction', createConversionFunction(...[])];
+    mappings['editor.action.previousMatchFindAction'] = ['monaco.editor.action.previousMatchFindAction', createConversionFunction(...[])];
+    mappings['editor.action.nextSelectionMatchFindAction'] = ['monaco.editor.action.nextSelectionMatchFindAction', createConversionFunction(...[])];
+    mappings['editor.action.previousSelectionMatchFindAction'] = ['monaco.editor.action.previousSelectionMatchFindAction', createConversionFunction(...[])];
+    mappings['editor.action.openLink'] = ['monaco.editor.action.openLink', createConversionFunction(...[])];
+    mappings['editor.action.quickFix'] = ['monaco.editor.action.quickFix', createConversionFunction(...[])];
+    mappings['editor.action.refactor'] = ['monaco.editor.action.refactor', createConversionFunction(...[])];
+    mappings['editor.action.sourceAction'] = ['monaco.editor.action.sourceAction', createConversionFunction(...[])];
+    mappings['editor.action.organizeImports'] = ['monaco.editor.action.organizeImports', createConversionFunction(...[])];
+    mappings['editor.action.autoFix'] = ['monaco.editor.action.autoFix', createConversionFunction(...[])];
+    mappings['editor.action.fixAll'] = ['monaco.editor.action.fixAll', createConversionFunction(...[])];
+    mappings['editor.action.rename'] = ['monaco.editor.action.rename', createConversionFunction(...[])];
```

Comment:
What about conversions between 0-based positions/locations and 1-based? This command will produce bogus results without conversions.

I thought this PR used to have generic conversion similar to what VS Code does: https://github.com/Microsoft/vscode/blob/70160d0e580d0588e37c93fe8724014d76b00939/src/vs/workbench/api/node/extHostCommands.ts#L110-L138

One can test it with extract refactorings for typescript vscode extension.

Reply:
No, the idea was always to explicitly convert. That way, it also works for stuff registered by clients that are not targeted to monaco.
> // automagically convert some argument types

automagically: brr.....

Reply:
It is just we need to review then all mapped commands and make sure that conversions are proper. Should it be a part of this PR?

Reply:
Well, "automagic" conversion needs to be tested, as well. I believe all the monaco action need the same conversions (0- vs 1-based). Problem is: we're not using the monaco location conventions everywhere in Theia, so I don't think we can generally convert parameters.
What about a plugin registering a command that is invoked by another plugin (with positional parameters). All those considerations led me to decide against the automatic conversion. I know it's more code, but safer, IMO.

Reply:
> What about a plugin registering a command that is invoked by another plugin (with positional parameters).

We could apply it only if a command is mapped to monaco command, i.e. we define a single monaco command conversion function and pass it as a second argument: `['monaco.editor.action.rename', monacoConversion]`. We should also not forget to convert a result.